src/main/resources/avro/common.avdl

```diff
@@ -17,31 +29,88 @@ error GAException {
 }
 
 /**
-An abstraction for referring to a genomic position, in relation to some
-already known reference. For now, represents a genomic position as a reference
-name, a base number on that reference (0-based), and a flag to say if it's the
-forward or reverse strand that we're talking about.
+Each base in a sequence graph has two sides, `PLUS` and `MINUS`.
+
+`PLUS` represents forwards, or the direction of increasing coordinates, while
+`MINUS` represents reverse-complement, or the direction of decreasing
+coordinates.
+*/
+enum Side {
+  PLUS,
+  MINUS
+}
+
+/**
+An abstraction for referring to a genomic position on a sequence.
+
+For now, represents a genomic position as a sequence ID, a base offset on that
+sequence (0-based), and a flag to say if it's the forward or reverse strand that
+we're talking about.
 */
-// TODO: Add support here for universally unique base IDs.
 record Position {
   /**
-  The name of the reference (or, more technically, the scaffold) in whatever
-  reference set is being used. Does not generally include a "chr" prefix, so for
-  example "X" would be used for the X chromosome.
+  The ID of the sequence on which the `Position` is located.
   */
-  string referenceName;
+  string id;
```

Comment:
`Position`s absolutely need to have IDs for the sequences they are on. Otherwise you would just have an offset and a side, without knowing what piece of sequence that offset was supposed to be into. The `Segment` object doesn't actually itself store the ID of the sequence it is talking about; it relies on its start `Position` to hold it.

Is backwards compatibility a concern for us beyond how much work it takes to write code for the new version? I was assuming we would do what most web APIs do and have a server present each supported version of the API behind a different endpoint: you would have http://ga4gh.org/api/v0.5/referencesets/{id} and http://ga4gh.org/api/v0.6/referencesets/{id} and so on for each version of the API that your server supports.

If we want to have clients that expect the old API hitting against servers that present the new API, they'll choke as soon as they encounter a `GraphAlignment` on the reads side, and the Variants side has been substantially redesigned. Backwards compatibility on the References I think is also broken by the refactoring I did to make a `Reference` use a `Segment`, so I'm not sure that keeping compatibility at the level of `Position`s is really worth it.

Would we prefer a sort of hybrid API where all the old endpoints and objects are available, but there are new versions of everything under different names?